File: app/frontend/lib/models/student.dart
import 'package:flutter/material.dart';

class Student {
  final int? studentId; // 학생 ID
  final String name;
  final String? grade; // 성인일 경우에만 입력
  final List<String> subjects;
  final String phone;
  final int sessions;
  final int completedSessions;
  final Color color;
  final String nextClass; // ex) "11월 7일 10:00"
  final int attendanceRate; // 0~100
  final bool isAdult; // 성인 여부 (기본값: true)
  final bool isActive; // 활성화 여부 (기본값: true)

  Student({
    this.studentId,
    required this.name,
    this.grade,
    required this.subjects,
    required this.phone,
    required this.sessions,
    required this.completedSessions,
    required this.color,
    required this.nextClass,
    required this.attendanceRate,
    this.isAdult = true, // 디폴트는 성인
    this.isActive = true, // 디폴트는 활성화
  });
}

File: app/backend/schemas/student.py
# app/backend/schemas/student.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime, date

class StudentBase(BaseModel):
    name: str
    phone: str
    parent_phone: Optional[str] = None
    teacher_id: Optional[int] = None
    school: Optional[str] = None
    grade: Optional[str] = None
    subject: Optional[str] = None
    start_date: Optional[date] = None
    lesson_day: Optional[str] = None
    lesson_time: Optional[str] = None
    hourly_rate: Optional[int] = None
    notes: Optional[str] = None
    is_active: Optional[bool] = True
    is_adult: Optional[bool] = False

class StudentCreate(StudentBase):
    pass

class StudentUpdate(BaseModel):
    name: Optional[str] = None
    phone: Optional[str] = None
    parent_phone: Optional[str] = None
    teacher_id: Optional[int] = None
    school: Optional[str] = None
    grade: Optional[str] = None
    subject: Optional[str] = None
    start_date: Optional[date] = None
    lesson_day: Optional[str] = None
    lesson_time: Optional[str] = None
    hourly_rate: Optional[int] = None
    notes: Optional[str] = None
    is_active: Optional[bool] = None
    is_adult: Optional[bool] = None

class StudentOut(StudentBase):
    model_config = ConfigDict(from_attributes=True)
    student_id: int
    created_at: datetime
    updated_at: datetime

class StudentListResp(BaseModel):
    total: int
    page: int
    pageSize: int
    items: list[StudentOut]
